### PR TITLE
Here you go James

### PR DESCRIPTION
**Pull request by Noviny**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -288,17 +314,14 @@ converters.ImportSpecifier = (path, context) => {
       };
     }
 
-    return convert(exported, {
-      ...context,
```
Comment (thejameskyle) [inline index.js]: Why can't you use object spread?

File: index.js [hunk 2]
```diff
@@ -30,14 +29,40 @@ converters.Program = (path, context) => {
...
+converters.ObjectProperty = (path, context) => {
+  const key = convert(path.key, context)
+  const value = generate(path.value).code;
+  return { key, value, kind: path.value.type }
+}
+
+converters.ObjectMethod = (path, context) => {
+  const key = convert(path.key, context)
+  const value = generate(path.body).code;
+  return { key, value, kind: path.body.type }
```
Comment (thejameskyle) [inline index.js]: Can you manually specify the different `kind`'s? I don't want to reuse the Babel ones in some places and not others

File: index.js [hunk 3]
```diff
@@ -72,20 +97,23 @@ converters.GenericTypeAnnotation = (path, context) => {
...
-  if (kind === 'reference') {
-    let bindingPath;
-
-    if (isFlowIdentifier(path)) {
-      let flowBinding = findFlowBinding(path, path.node.name);
-      if (!flowBinding) throw new Error;
-      bindingPath = flowBinding.path.parentPath;
-    } else {
-      bindingPath = path.scope.getBinding(path.node.name);
+  if (context.mode === 'value') return path.name;
```
Comment (thejameskyle) [inline index.js]:
```js
return { kind: 'id', name: path.node.name };
```

File: index.js [hunk 4]
```diff
@@ -72,20 +97,23 @@ converters.GenericTypeAnnotation = (path, context) => {
...
+  if (context.mode === 'type') {
```
Comment (thejameskyle) [inline index.js]: else if

File: index.js [hunk 5]
```diff
@@ -30,14 +29,40 @@ converters.Program = (path, context) => {
...
+      result.classes.push(classProperties);
     },
   });
-
   return result;
 };
 
+converters.ObjectProperty = (path, context) => {
+  const key = convert(path.key, context)
+  const value = generate(path.value).code;
```
Comment (thejameskyle) [inline index.js]: Don't use `generate`

Comment (Noviny) [reply index.js]: What's the best alternative to generate for what I want?

File: index.js (hunk 1 above)
Comment (Noviny) [reply index.js]: Tests were failing on using spread. Not sure of why.

File: index.js (hunk 5 above)
Comment (thejameskyle) [reply index.js]: Keep `convert()`-ing child nodes until you have an AST for the value you care about.

**Comment by Noviny** [comment]
Going to create a new branch for this after rebasing.

**Pull request closed without merging**